Bug 1486869 - Make Clear Private Data settings panel more granular

mrachamallu opened the pull request:
*Thank you for submitting a pull request, your contributions are greatly appreciated!*

Added a new option in settings to manage data for specific website.

## Pull Request Checklist

- [ ] My patch has gone through review and I have addressed review comments
- [ ] My patch has a standard commit message that looks like `Bug 12345678 - This fixes something something`

- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have marked the bug with `[needsuplift]`
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here.

## Notes for testing this patch

If useful, please leave notes for QA, explaining what this patch changes and how it can be best tested and verified.

farhanpatel commented:
This looks really good Meera! I'm still going through it but one thing I noticed 
- remove a website via the search
- press cancel
- the website still appears in the list 

It looks like the website data is not refreshed after leaving a search

mrachamallu commented:
Thanks for the comments, Farhan
-I had implemented a function to reload data after pressing cancel on search and the websites I delete disappear from the list, not sure why it's not working for you, but I'll look into it again
-I changed WebsiteDataManagementViewController.swift to enum style. It looks a little different than ActivityStreamPanel.swift, but if it's all good, then I'll change the other two tableviews to this style as well

farhanpatel commented on Client/Frontend/Settings/websiteSearchResultsViewController.swift:
```diff
@@ -0,0 +1,136 @@
+    let flexible = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.flexibleSpace, target: self, action: nil)
+    let editButton: UIBarButtonItem = UIBarButtonItem(title: "Edit", style: .plain, target: self, action: #selector(didPressEdit))
+    let doneButton: UIBarButtonItem = UIBarButtonItem(title: "Done", style: .done, target: self, action: #selector(didPressDone))
+    let deleteButton: UIBarButtonItem = UIBarButtonItem(title: "Delete", style: .plain, target: self, action: #selector(didPressDelete))
```
These buttons need localized strings in Strings.swift

farhanpatel commented on Client/Frontend/Settings/WebsiteDataManagementViewController.swift:
```diff
@@ -0,0 +1,196 @@
+            let site = siteRecords[indexPath.item]
+            cell.textLabel?.text = site.nameOfSite
+        case .showMore:
+            cell.textLabel?.text = "Show More"
```
Needs a localized string

farhanpatel commented on Client/Frontend/Settings/WebsiteDataManagementViewController.swift:
```diff
@@ -0,0 +1,196 @@
+        case .showMore:
+            getAllWebsiteData()
+        case .button:
+            func clearwebsitedata(_ action: UIAlertAction) {
```
Move the method outside of the `didSelectRowAt` it'll make it cleaner to understand

farhanpatel commented on Client/Frontend/Settings/WebsiteDataManagementViewController.swift:
```diff
@@ -0,0 +1,196 @@
+                tableView.reloadData()
+            }
+            let alert =  UIAlertController.clearWebsiteDataAlert(okayCallback: clearwebsitedata)
+            let generator = UIImpactFeedbackGenerator(style: .heavy)
```
Nice 😄

farhanpatel commented on Client/Frontend/Settings/ClearPrivateDataTableViewController.swift:
```diff
@@ -100,76 +105,91 @@ class ClearPrivateDataTableViewController: ThemedTableViewController {
+            let view = WebsiteDataManagementViewController()
+            navigationController?.pushViewController(view, animated: true)
+        } else if indexPath.section == SectionButton {
+            func clearPrivateData(_ action: UIAlertAction) {
```
Moving this method outside of `didSelectRowAt` will clean this up nicely

mrachamallu commented on Client/Frontend/Settings/WebsiteDataManagementViewController.swift:
```diff
@@ -0,0 +1,196 @@
+        searchController.searchBar.placeholder = "Filter Sites"
+        searchController.searchBar.delegate = self
+        if #available(iOS 11.0, *) {
+            navigationItem.searchController = searchController
```
updated the else statement so searchController is added to the titleView
